### PR TITLE
Do not initialize performance if required apis are not available

### PR DESCRIPTION
**Pull request by alikn**
Context: https://github.com/firebase/firebase-js-sdk/issues/1774

### CODE REVIEW

File: packages/performance/src/services/perf_logger.ts [hunk 1]
```diff
@@ -107,6 +107,10 @@ export function logTrace(trace: Trace): void {
   if (!settingsService.dataCollectionEnabled && !trace.isAuto) {
     return;
   }
+  // Do not log if required apis are not available.
+  if (!Api.getInstance().requiredApisAvailable()) {
```
Comment (zijianjoy) [inline packages/performance/src/services/perf_logger.ts]: Should we also add these "if condition" to logNetworkRequest() in this file?

File: packages/performance/src/controllers/perf.test.ts [hunk 2]
```diff
@@ -38,6 +42,16 @@ describe('Firebase Performance Test', () => {
     options: fakeFirebaseConfig
   } as unknown) as FirebaseApp;
 
+  describe('#constructor', () => {
+    it('does not initialize performance if the required apis are not available', () => {
+      stub(Api.prototype, 'requiredApisAvailable').returns(false);
+      stub(initializationService, 'getInitializationPromise');
+      new PerformanceController(fakeFirebaseApp);
+
+      expect(initializationService.getInitializationPromise).not.be.called;
```
Comment (zijianjoy) [inline packages/performance/src/controllers/perf.test.ts]: Add a check on the existence of console log?

For example:
expect( console.log.calledWith('Firebase Performance cannot start if browser does not support fetch and Promise or cookie is disabled.') ).to.be.true;

File: packages/performance/src/services/perf_logger.ts (hunk 1 above)
Comment (alikn) [reply packages/performance/src/services/perf_logger.ts]: No need, the trace api can be used to create custom trace even if the sdk is not initialized, but there is no public api for networks.

File: packages/performance/src/controllers/perf.test.ts (hunk 2 above)
Comment (alikn) [reply packages/performance/src/controllers/perf.test.ts]: Added a check for the consoleLogger to be called. Not sure about adding the whole message though.

File: packages/performance/src/services/perf_logger.ts (hunk 1 above)
Comment (zijianjoy) [reply packages/performance/src/services/perf_logger.ts]: Got it, suggest to use console logger to log that required APIs are not available for the scenario above.

**Pull request merged**